frontend/z3_axioms.py

```diff
@@ -40,19 +38,10 @@ def mult(left, right, result, types):
     """
     return [
         Or(
-            And(types.subtype(left, types.int), types.subtype(right, types.seq), result == right),
             And(types.subtype(left, types.seq), types.subtype(right, types.int), result == left),
-            And(types.subtype(left, types.num), types.subtype(right, types.num), types.subtype(result, types.num))
-        ),
-
-        Implies(
-            And(types.subtype(left, types.num), types.subtype(right, types.num), types.stronger_num(left, right)),
-            result == left
-        ),
-
-        Implies(
-            And(types.subtype(left, types.num), types.subtype(right, types.num), types.stronger_num(right, left)),
-            result == right
+            And(types.subtype(left, types.int), types.subtype(right, types.seq), result == right),
+            And(types.subtype(left, types.num), types.subtype(right, left), result == left),
+            And(types.subtype(left, types.num), types.subtype(left, right), result == right),
```

Comment:
Same here: And(types.subtype(right, types.num), types.subtype(left, right), result == right)